dds/DCPS/RTPS/Util.cpp

```diff
@@ -0,0 +1,38 @@
+#include "Util.h"
+
+#include "dds/DCPS/Definitions.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace RTPS {
+
+bool bitmapNonEmpty(const OpenDDS::RTPS::SequenceNumberSet& snSet)
+{
+  const size_t num_ulongs = (snSet.numBits + 31) / 32;
+
+  OPENDDS_ASSERT(num_ulongs <= snSet.bitmap.length());
+
+  if (num_ulongs == 0) {
+    return false;
+  }
+
+  const size_t last_index = num_ulongs - 1;
+  if (last_index > 0) {
+    for (CORBA::ULong i = 0; i < last_index; ++i) {
+      if (snSet.bitmap[i]) {
+        return true;
+      }
+    }
+  }
+
+  const CORBA::ULong mod = snSet.numBits % 32;
+  const CORBA::ULong mask = mod ? ~((0x1u << (32 - mod)) - 1) : 0xFFFFFFFF;
```

Comment:
```suggestion
  const CORBA::ULong mask = mod ? -(1u << (32 - mod)) : 0xFFFFFFFF;
```
I think this gives the same values...

Reply:
On my build, the generated assembly winds up being the same, but it's closer to your version so I'll change my code over.

Reply:
Right, I don't know if one is clearly better than the other but I guess I'm leaning towards fewer operators (especially if we handle `mod==0` somewhere else).